src/sysupdate/sysupdate-resource.h

```diff
@@ -66,12 +66,24 @@ static inline bool RESOURCE_IS_URL(ResourceType t) {
                       RESOURCE_URL_FILE);
 }
 
+typedef enum PathRelativeTo {
+        /* Please make sure to folow the naming of the corresponding PartitionDesignator enum values,
+         * where this makes sense, like for the following three. */
+        PATH_RELATIVE_TO_ROOT,
+        PATH_RELATIVE_TO_ESP,
+        PATH_RELATIVE_TO_XBOOTLDR,
+        PATH_RELATIVE_TO_BOOT, /* Refers to $BOOT from the BLS. No direct counterpart in PartitionDesignator */
+        _PATH_RELATIVE_TO_MAX,
+        _PATH_RELATIVE_TO_INVALID = -1,
```

Comment:
oh, actually our newest coding style suggests we use `-EINVAL` instead of `-1` for this